x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/fields/field_types/geo_shape_type.tsx

```diff
@@ -0,0 +1,145 @@
+    }
+    default:
+      return false;
+  }
+};
+
+interface Props {
+  field: NormalizedField;
+}
+
+export const GeoShapeType = ({ field }: Props) => {
+  return (
+    <>
+      <AdvancedSettingsWrapper>
+        <EuiCallOut color="primary">
+          <p>
+            <FormattedMessage
+              id="xpack.idxMgmt.mappingsEditor.geoShape.infoMessage"
+              defaultMessage="GeoShape types are indexed by decomposing the shape into a triangular mesh and indexing each triangle as a 7 dimension point in a BKD tree. {docsLink}"
+              values={{
+                docsLink: (
+                  <EuiLink
+                    href={documentationService.getGeoShapeDocumentationLink()}
+                    target="_blank"
+                  >
+                    {i18n.translate('xpack.idxMgmt.mappingsEditor.geoShape.learnMoreLink', {
+                      defaultMessage: 'Learn more.',
+                    })}
+                  </EuiLink>
+                ),
+              }}
+            />
+          </p>
+        </EuiCallOut>
+
+        <EuiSpacer />
+
+        <EditFieldSection>
+          {/* orientation */}
+          <EditFieldFormRow
+            title={
+              <h3>
+                {i18n.translate('xpack.idxMgmt.mappingsEditor.orientationFieldTitle', {
+                  defaultMessage: 'Set orientation',
+                })}
+              </h3>
+            }
+            description={i18n.translate(
+              'xpack.idxMgmt.mappingsEditor.geoShape.orientationFieldDescription',
+              {
+                defaultMessage:
+                  'Define how to interpret vertex order for polygons or multipolygons',
+              }
+            )}
+            toggleDefaultValue={getDefaultValueToggle('orientation', field.source)}
+          >
+            <UseField
+              path="orientation"
+              config={getFieldConfig('orientation')}
+              component={SelectField}
+              componentProps={{
+                euiFieldProps: {
+                  options: PARAMETERS_OPTIONS.orientation,
+                  style: { maxWidth: 300 },
+                },
+              }}
+            />
+          </EditFieldFormRow>
+
+          {/* points_only */}
+          <EditFieldFormRow
+            title={
+              <h3>
+                {i18n.translate('xpack.idxMgmt.mappingsEditor.geoShape.pointsOnlyFieldTitle', {
+                  defaultMessage: 'Points only',
+                })}
+              </h3>
+            }
+            description={i18n.translate(
+              'xpack.idxMgmt.mappingsEditor.geoShape.ignoreZValueFieldDescription',
+              {
+                defaultMessage: 'Configures the geo_shape field type for point shapes only.',
+              }
+            )}
+            formFieldPath="points_only"
+          />
+
+          {/* ignore_z_value */}
+          <EditFieldFormRow
```

Comment:
I think this is a good candidate for a separate component as it is used in 3 places.